src/controllers/request-controller.ts

```diff
@@ -117,46 +66,47 @@ export default class RequestController {
       const { cid, docId } = req.body;
 
       if (cid == null) {
-        return res.status(BAD_REQUEST).send({
+        return res.status(StatusCodes.BAD_REQUEST).send({
           error: 'CID is empty',
         });
       }
 
       if (docId == null) {
-        return res.status(BAD_REQUEST).send({
+        return res.status(StatusCodes.BAD_REQUEST).send({
           error: 'Document ID is empty',
         });
       }
 
       const cidObj = new CID(cid);
-      let request: Request = await this.requestRepository.findByCid(cidObj);
-      if (request != null) {
-        return res.status(BAD_REQUEST).send('CID has already been submitted');
+      let request = await this.requestRepository.findByCid(cidObj);
+      if (request) {
+        const body = await this.#requestPresentation.body(request);
+        return res.status(StatusCodes.ACCEPTED).json(body);
+      } else {
+        request = new Request();
+        request.cid = cid.toString();
+        request.docId = docId;
+        request.status = RequestStatus.PENDING;
+        request.message = 'Request is pending.';
+
+        request = await this.requestRepository.createOrUpdate(request);
+
+        const cron = awsCronParser.parse(config.cronExpression);
+
+        return res.status(StatusCodes.CREATED).json({
```

Comment:
might as well move this case into requestPresentation and use it here, to keep everything in the same place.

Reply:
whoops, true